pandas/core/algorithms.py

```diff
@@ -988,7 +988,9 @@ def _get_data_algo(values, func_map):
     elif is_unsigned_integer_dtype(values):
         f = func_map['uint64']
         values = _ensure_uint64(values)
-
+    elif is_categorical(values) and values._ordered:
```

Comment:
no, don't use a private ``values._ordered`` here. and use ``is_categorical_dtype``